### PR TITLE
readline8,bash5: init

### PR DESCRIPTION
**Pull request by dtzWill**
###### Motivation for this change

New bash (and readline)! \o/

This adds the new versions but
doesn't change the defaults for anything.

"yet". :)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Tested briefly, and with x86_64 musl.

### CODE REVIEW

**Comment by dtzWill** [comment]
Statically linked bash 5 using musl: `/nix/store/9wfgiygqnzzfz6fpd5jsbr8fifawr3d3-bash-5.0-p0` which you can grab via the ALLVM cachix FWIW!  See https://allvm.cachix.org for setup, then point `nix-store -r` or other commands at the path.

Built without explicitly disabling bash malloc, which might be a problem but thought I'd see how vanilla bash worked :).  And recent musl's support for malloc interposition/replacement means there's reason for it to maybe work.  Anyway, enjoy!

**Comment by Mic92** [comment]
Why not upgrading our own bash version instead?

**Review by Ma27 — APPROVED** [review]

**Comment by dtzWill** [comment]
I wanted to try it without certifying it doesn't break things ;).

I imagine it's relatively safe?

Certainly not a change to be made lightly, IMO.

Behavioral differences that are subtle might break things
not immediately apparent, for example this in the announcement:

> There are a number of changes to the
> expansion of $@ and $* in various contexts where word splitting is not
> performed to conform to a Posix standard interpretation, and additional
> changes to resolve corner cases for Posix conformance.

File: pkgs/shells/bash/5.0.nix [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+      functional improvements over sh for both programming and
+      interactive use.  In addition, most sh scripts can be run by
+      Bash without modification.
+    '';
+
+    license = licenses.gpl3Plus;
+
+    platforms = platforms.all;
+
+    maintainers = [ maintainers.peti ];
```
Comment (Infinisil) [inline pkgs/shells/bash/5.0.nix]: You might want to have your own name here instead

**Comment by Mic92** [comment]
mhm for readline it already seems to be too late to switch conventions in naming. Let's not make the same mistake for bash.

**Pull request merged**